packages/macro/src/index.macro.ts

```diff
@@ -37,6 +37,39 @@ interface PluginOptions {
   };
 }
 
+export type Options = {
+  webpackChunkName?: string;
+}
+
+type PrimitiveObjectProperty = t.ObjectProperty & {
+  key: t.Identifier | t.StringLiteral;
+  value: t.StringLiteral | t.BooleanLiteral | t.NumericLiteral;
+}
+
+function isPrimitiveObjectProperty(property: t.ObjectProperty | t.ObjectMethod | t.SpreadElement): property is PrimitiveObjectProperty {
+  return t.isObjectProperty(property) &&
+         (t.isIdentifier(property.key) || t.isStringLiteral(property.key)) &&
+         (t.isStringLiteral(property.value) || t.isBooleanLiteral(property.value) || t.isNumericLiteral(property.value));
+}
+
+function parseToOptions(optionsNode: t.ObjectExpression): Options {
+  if (optionsNode.properties.length === 0) {
+    return {};
+  }
+
+  const options = {};
+  for (const property of optionsNode.properties) {
+    if (!isPrimitiveObjectProperty(property)) {
+      throw new Error("Options object can only contain properties with primitive values.")
+    }
+
+    const key = t.isStringLiteral(property.key) ? property.key.value : property.key.name;
+    options[key] = property.value.value;
```

Comment:
I ensured that the options object is having primitive properties (not variable or complex ones). I didn't ensure that the object is having only `webpackChunkName` property for forward compatibility.